tests/MainTest.elm

```diff
@@ -68,65 +53,232 @@ suite =
         , Test.fuzz choiceStringFuzzer "choiceStringToChoice,choiceStringFromString is reversible" <|
             \choice ->
                 choice
-                    |> encodeChoice
+                    |> Main.encodeChoice
                     |> Json.Encode.encode 0
-                    |> Json.Decode.decodeString decodeChoice
+                    |> Json.Decode.decodeString Main.decodeChoice
                     |> Expect.equal (Ok choice)
+        , Test.fuzz pairOfFormFieldFuzzer "encode old FormField can be decoded with decodeFormField" <|
+            \{ oldField, newField } ->
+                oldField
+                    |> encodeFormField
+                    |> Json.Encode.encode 0
+                    |> Json.Decode.decodeString Main.decodeFormField
+                    |> Expect.equal (Ok newField)
```

Comment:
finally, we test
1. encoding the old form field value into json
2. decoding the json as new form field
3. assert value is expected